great_expectations/types/__init__.py

```diff
@@ -188,12 +188,14 @@ def property_names(
 
         # Gather private fields:
         # By Python convention, properties of non-trivial length, prefixed by underscore ("_") character, are private.
-        property_names: Set[str] = set(
+        private_fields: Set[str] = set(
             filter(
                 lambda name: len(name) > 1,
                 [key[1:] for key in self.keys() if key[0] == "_"],
             )
         )
+        public_fields: Set[str] = {key for key in self.keys() if key[0] != "_"}
```

Comment:
@kenwade4 Is the idea here that in some objects we do not have private fields only, but also public fields like `self.my_field = "hello"`?  If so, then I would like to propose that for the objects that extend `DictDot` we avoid this pattern and instead only have private fields and `@property` getter and setters.  In fact, putting a comment here into the present module saying that this is what subclasses must do would be extremely useful.  Might you be able to change this?  Thanks!

Reply:
The property_names set would only ever contain items from `self.keys()` if they started with an underscore. Everything else in this function just iterates over that property_names set via different comprehensions.

For any of the many objects we have that are like the following

```
@dataclass
class Whatever(SerializableDictDot):
    first: str
    second: Dict[str, Any]
    third: bool
```

... their `self.keys()` contains the properties defined (i.e. ["first", "second", "third"])... none of those start with an underscore and thus the property_names set would be empty and an empty set would be returned.

The `to_dict` method creates its internal new_dict by iterating over the keys returned by self.property_names(), so all those dataclasses just have an empty dict returned

Reply:
@kenwade4 Let's discuss -- this is important to get right.  Thanks!